### ISSUE TITLE
Good IOU Score on training data, but bad segmentation on testing data.

### ISSUE THREAD

**Issue by ranjan2601** [qubvel/segmentation_models#571]
I have used inceptionv3 as my backbone and trained it on 20 epochs, got mean iou score of 0.90. However, when i try to predict on test data, the results are like this:

![download](https://github.com/qubvel/segmentation_models/assets/82278266/90f4dcb2-5087-4d8e-b1b9-d11781f07385)

My dataset contains images about tomato disease(10 classes). The Gt mask tells which part on leaf is healthy and which part is diseased, whereas Pr mask is just making the shape of the leaf.
Thanks

**Comment by nataliameira** [comment]
Hello @ranjan2601,

Could you share how you organized your `class Dataset` and built your model?

It will be easier to come up with a strategy for your problem.
